src/constraints/constraint_operating_point_bounds.jl

```diff
@@ -20,29 +20,18 @@
 """
     add_constraint_operating_point_bounds!(m::Model)
 
-Limit the operating point flow variables `unit_flow_op` to the difference between successive operating points times
-the capacity of the unit.
+Limit the maximum number of each activated segment `unit_flow_op_active` cannot be higher than the number of online units.
 """
 function add_constraint_operating_point_bounds!(m::Model)
-    @fetch unit_flow_op, units_available = m.ext[:spineopt].variables
-    t0 = _analysis_time(m)
+    @fetch unit_flow_op_active, units_on = m.ext[:spineopt].variables
     m.ext[:spineopt].constraints[:operating_point_bounds] = Dict(
         (unit=u, node=n, direction=d, i=op, stochastic_scenario=s, t=t) => @constraint(
             m,
-            + unit_flow_op[u, n, d, op, s, t]
-            <=
-            (
-                + operating_points[(unit=u, node=n, direction=d, stochastic_scenario=s, analysis_time=t0, i=op)] - (
-                    (op > 1) ?
-                    operating_points[(unit=u, node=n, direction=d, stochastic_scenario=s, analysis_time=t0, i=op - 1)] :
-                    0
-                )
-            )
-            * unit_capacity[(unit=u, node=n, direction=d, stochastic_scenario=s, analysis_time=t0, t=t)]
-            * units_available[u, s, t]
-            * unit_conv_cap_to_flow[(unit=u, node=n, direction=d, stochastic_scenario=s, analysis_time=t0, t=t)]
-            # TODO: extend to investment functionality ? (is that even possible)
-        ) for (u, n, d) in indices(unit_capacity)
-        for (u, n, d, op, s, t) in unit_flow_op_indices(m; unit=u, node=n, direction=d)
+            unit_flow_op_active[u, n, d, op, s, t]
+            <= 
+            units_on[u, s, t]
+        )
+        for (u, s, t) in units_on_indices(m)
+        for (u, n, d, op, s, t) in unit_flow_op_indices(m; unit=u, stochastic_scenario=s, t=t)
```

Comment:
The stochastic path of this constraint would break as soon as units on and unit flow would have different stochastic scenarios. The only solution to this at the moment would probably be to use a designated ` constraint_operating_point_bound_indices` function and provide stochastic paths for both variables

Reply:
One thing to keep in mind is how we would constrain e.g. different fuels in these settings. I guess the only way at this point would be to model this via 2 units with a designated user constraints. Alternatively we could also think of a constraint in the form of (draft example):
```julia
sum(unit_flow_op_active(u,n_i,d_i,op,s,t) for (u,n_i,d_i,op,s,t) in
unit_flow_op_indices(unit=u,op=op,s=s,t=t)) <= 1 forall (u,op,s,t)
```
This could be an optional constraint to have a binary choice of which fuels and corresponding heat rates are used.

Reply:
Note that this would only work for binary online_variable_types

Reply:
Does the `constraint_unit_flow_capacity_indices` provide the needed indices?

Reply:
I tested it a bit by giving `unit_on` a different stochastic scenario to the unit and the system does not break.

In what case would the `unit_flow` and `unit_on` use different stochastic scenarios? A dedicated stochastic scenario for a unit is set for `unit_on`. Isn't `unit_flow` always bonded with the stochastic scenario used by the `unit`? I'm actually not clear on how the stochastic scenario feature is implemented (e.g. how `constraint_operating_point_bound_indices` should look like?) Would extremely appreciate any further explanation!

Reply:
Commit 13e6f70b7b23c3fe44f2027180a47e7b22558daf attempts to resolve the first concern on constraint indices thanks @manuelma for your help!

Reply:
Improvement on this feature will continue in issue #631.

> One thing to keep in mind is how we would constrain e.g. different fuels in these settings. I guess the only way at this point would be to model this via 2 units with a designated user constraints. Alternatively we could also think of a constraint in the form of (draft example):
> 
> ```julia
> sum(unit_flow_op_active(u,n_i,d_i,op,s,t) for (u,n_i,d_i,op,s,t) in
> unit_flow_op_indices(unit=u,op=op,s=s,t=t)) <= 1 forall (u,op,s,t)
> ```
> 
> This could be an optional constraint to have a binary choice of which fuels and corresponding heat rates are used.